pepper-apis/docs/specification/src/components/request-bodies.yml

```diff
@@ -170,3 +170,15 @@ JoinMailingListPayload:
     application/json:
       schema:
         $ref: '../pepper.yml#/components/schemas/JoinMailingListPayload'
+UpdateUserLastVisitedSectionRequestBody:
+  description: to update user's last visited section
+  required: true
+  content:
+    application/json:
+      schema:
+        type: object
+        required:
+          - lastVisitedSection
```

Comment:
Since we don't have section identifiers, we are probably referring to the sectional ordinal index. Maybe we should call it `lastVisitedSectionIndex` and document that it starts at 0? Or 1?

Reply:
Also, if the user is viewing, say, a four-part activity, the first time they go up to section 3.
Then the next day they come back, load the activity, and hit the next button up to section 2.
What do we want this value to show?

Reply:
If we are going to use PATCH to URL that includes /last-visited-section, then including the lastVisitedSection property in body seem redundant.
Might be cleaner to PATCH to the activity instance URL without the /last-visited part in the path. Leave door open to PATCHing other properties in future as well.

Reply:
> Since we don't have section identifiers, we are probably referring to the sectional ordinal index. Maybe we should call it `lastVisitedSectionIndex` and document that it starts at 0? Or 1?

will do

 

> Also, if the user is viewing, say, a four-part activity, the first time they go up to section 3.
> Then the next day they come back, load the activity, and hit the next button up to section 2.
> What do we want this value to show?

depends on what the client sends. In the current task, we need to track the progress of the activity, so we'll update the parameter for each step forward between sections(next button). 


> If we are going to use PATCH to URL that includes /last-visited-section, then including the lastVisitedSection property in body seem redundant.
> Might be cleaner to PATCH to the activity instance URL without the /last-visited part in the path. Leave door open to PATCHing other properties in future as well.

will change the name of the endpoint to `/user/{user}/studies/{study}/activities/{instance}/section:`